content/blog/2022-05-17-aws-remotes-in-dvc.md

```diff
@@ -0,0 +1,138 @@
+with DVC commands. We’ll be working with
+[this repo](https://github.com/iterative/stale-model-example) if you want an
+example to play with. By the time you finish, you should be able to create this
+setup for any ML project using an AWS remote.
+
+## Set up an AWS S3 bucket
+
+Make sure that you already have an [AWS account](https://aws.amazon.com/) and
+log in. Search for `S3` and it should be the first service that appears.
+
+![S3 service in AWS](/uploads/images/2022-05-17/finding_s3.png)
+
+Once you’re on the S3 page, click the `Create Bucket` button and it will take
+you to a page that looks like this. The bucket in this example is called
+`updatedbikedata` because that is the data our demo repo works with. You can
+leave the default settings in place or you can update them to fit the
+functionality you need.
+
+![create an S3 bucket in AWS](/uploads/images/2022-05-17/create_bucket.png)
+
+Once you’ve created the bucket, you should be redirected to the S3 dashboard and
+see the success message and your new bucket.
+
+![newly created S3 bucket in AWS](/uploads/images/2022-05-17/created_bucket.png)
+
+### Get your credentials
+
+Now that the S3 bucket is ready, we need the `access_key_id` and
+`secret_access_key` from AWS in order to connect to our project. You can create
+these keys in your Identity and Access Management settings. Go to your security
+credentials and select the `Access keys` section. Then click the
+`Create New Access Key` button. This will generate a new set of keys for you so
+make sure you download this file to get your secret access key.
+
+![make AWS access credentials](/uploads/images/2022-05-17/make_credentials.png)
+
+Once you’ve downloaded the credentials, you should see the access key ID in the
+table. Note that you won’t be able to access your secret key again at this
+point. You would need to make a new set of credentials if you don’t have it.
+
+![successfully created AWS access credentials](/uploads/images/2022-05-17/credentials.png)
+
+That’s it for setting up your bucket and getting the credentials you need! Now
+let’s add DVC to our demo repo and set up the remote.
+
+## Set up a DVC project
+
+First, add DVC as a requirement to your project with the following installation
+command:
+
+`$ pip install 'dvc[s3]'`
+
+Then you can initialize DVC in your own project with the following command:
+
+`$ dvc init`
+
+This will add all of the DVC internals needed to start versioning your data and
+tracking experiments. Now we need to set up the remote to connect our project
+data stored in AWS to the DVC repo.
+
+### Create a default remote
+
+Now we can add a default to the project with the following command:
+
+`$ dvc remote add -d bikes s3://updatedbikedata/train/`
+
+This creates a default remote called `bikes` that connects to the
+`updatedbikedata` bucket we made earlier and it references a folder inside this
+bucket called `train` which is where the training data for the model will be
+stored.
```

Comment:
Is there a reason to use a folder inside the bucket instead of the bucket directly? 

Could be that beginners might get confused thinking that they need to create a folder inside the bucket?

Reply:
That's a good point. I'll update this to just upload to the bucket directly.